modules/rostests/apitests/advapi32/Hash.c

```diff
@@ -0,0 +1,296 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for basic hash APIs
+ * COPYRIGHT:   Copyright 2023 Ratin Gao <ratin@knsoft.org>
+ */
+
+#include "precomp.h"
+
+typedef struct _A_SHA_CTX
+{
+    UCHAR Buffer[64];
+    ULONG State[5];
+    ULONG Count[2];
+} A_SHA_CTX, *PA_SHA_CTX;
+
+#define A_SHA_DIGEST_LEN 20
+
+typedef struct _MD5_CTX
+{
+    ULONG Count[2];
+    ULONG State[4];
+    UCHAR Buffer[64];
+    UCHAR Hash[16];
+} MD5_CTX, *PMD5_CTX;
+
+#define MD5_DIGEST_LEN 16
+
+typedef struct _MD4_CTX
+{
+    ULONG State[4];
+    ULONG Count[2];
+    UCHAR Buffer[64];
```

Comment:
It's quite interesting to observe that the Count and State arrays are reversed compared to the MD5 context...

Reply:
Extra question : This check for the layout of the context structures, is compared with what's done in Windows?
(I'm asking because our cryptlib code uses similar-named context structures but I wonder about their layout)

Reply:
Yes, it's interesting. The layout definitions in apitest is the result I observe the context data in Windows byte-to-byte.